lib/spack/spack/environment/environment.py

```diff
@@ -1515,7 +1515,6 @@ def install_specs(self, specs=None, **install_args):
 
         if not specs_to_install:
             tty.msg('All of the packages are already installed')
-            return
```

Comment:
What do you think of adding:
```python
with self.write_transaction():
    self.regenerate_views()
```
before the return statement instead of deleting it? This would be slightly better on debug, as it avoids messages like:
```
==> Processing 0 uninstalled specs
```

Reply:
Let me just put that in else: instead, cause I have no clue what other code paths are missed cause of the early return